### ISSUE TITLE
Inconsistent package exports

### ISSUE THREAD

**Issue by pwasemsap** [SAP/ui5-webcomponents-react#6467]
### Is your feature request related to a problem?

Imports across all packages are inconsistent: 

```typescript
import '@ui5/webcomponents-react/dist/Assets'
import { ThemeProvider } from '@ui5/webcomponents-react'
import employeeIcon from '@ui5/webcomponents-icons/employee.js'
```

Sometimes the path includes the `dist` folder. 
Sometimes the imports must end with `.js`
Sometimes the import includes an upper case word.

### Describe the solution you'd like

Imports should not contain the `dist` folder.
Imports should not end with `.js` 
Import should always be all lower case words.

```typescript
import '@ui5/webcomponents-react/assets'
import { ThemeProvider } from '@ui5/webcomponents-react'
import employeeIcon from '@ui5/webcomponents-icons/employee'
```

This can easily achieved by proper `exports` in `package.json`:

```json
{
  "exports": {
    "./components/*": "./dist/components/*.js",
    "./hooks/*": "./dist/hooks/*.js",
    "./utils/*": "./dist/utils/*.js"
  }
}
```

### Describe alternatives you've considered

None

### Additional Context

The described solution matches the current state of the react / web eco system and ESM imports:

```typescript
import { someMethod } from 'my-dep'
import React, { useEffect } from 'react'
import Image from 'next/image'
import { defineConfig } from 'vite'
import eslint from '@eslint/js'
import stylistic from '@stylistic/eslint-plugin'
```

### Organization

SAP SE

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Hi @pwasemsap,

I understand that the different import styles can be confusing, so let me explain the reasoning behind them:

`import '@ui5/webcomponents-react/dist/Assets'`

This import brings in assets from both `ui5-webcomponents` and `ui5-webcomponents-react`. Since these assets are side-effect imports, we can't switch to a named or default import. Tree-shaking is often not enabled in development environments, and because we want to allow control over whether these imports are used, we use a deep import path here.

`import { ThemeProvider } from '@ui5/webcomponents-react'`

These imports come from our package (`ui5-webcomponents-react`). Since using named or default imports is common in React, we’ve decided to expose most of our public APIs and components this way.

`import employeeIcon from '@ui5/webcomponents-icons/employee.js'`

This is a typical `ui5-webcomponents` import. Specifically, icon imports can be used both as named imports or as side-effect imports (e.g., `import '@ui5/webcomponents-icons/employee.js'`). Most `ui5-webcomponents` imports, including components and icons, are side-effect imports.

To summarize, we don't plan to change the imports because in most cases, it would either introduce drawbacks or simply wouldn’t be possible. I hope this explanation clarifies things.

**Comment by pwasemsap** [comment]
@Lukas742 

**This not about default or named exports !!!**

Please read such a request carefully before closing it.

It is about having proper imports without the `.js` ending and without having `dist` in the path.

Bad:

```typescript
import '@ui5/webcomponents-react/dist/Assets'
import employeeIcon from '@ui5/webcomponents-icons/employee.js'
```

Good:

```typescript
import '@ui5/webcomponents-react/assets'
import employeeIcon from '@ui5/webcomponents-icons/employee.'
```

This can be easily achieved by having proper `exports` in package.json as already mentioned above.

```json
{
  "exports": {
    "./*": "./dist/*.js"
  }
}
```

Having the "polluted" exports results in decreased developer experience and bad acceptance by developers.

Package consumers don't want to deal with an internal `dist` folder or file ending.

A clean import looks like this:

```typescript
import { someMethod } from 'my-dep'
import React, { useEffect } from 'react'
import Image from 'next/image'
import { defineConfig } from 'vite'
import eslint from '@eslint/js'
import stylistic from '@stylistic/eslint-plugin'
```

**Comment by Lukas742** [comment]
@pwasemsap We're using export paths for[ __our__ exports](https://github.com/SAP/ui5-webcomponents-react/blob/main/packages/main/package.json#L9-L26). Adding a dedicated export path for the `Assets` was something we considered, but decided not to offer, to follow the implementation of ui5-webcomponents. 

> Imports should not end with .js

Why? Also this is something you should be able to configure in your environment as long as it's correctly configured to resolve module extensions: https://stackblitz.com/edit/github-mars4f?file=src%2Fmain.tsx

We have no control over imports coming from [ui5-webcomponents](https://github.com/SAP/ui5-webcomponents) (e.g. `import employeeIcon from '@ui5/webcomponents-icons/employee.'`) and don't have plans to wrap or proxy them.
